### PR TITLE
Do not hide view on scroll events if scrollTop hasn't been changed

### LINKED ISSUE #305: [Firefox] Menu doesn't show up for textarea when parent has display: table and textarea is scrolled down a bit
~~Run the following code with the console in `index.html` on master or http://ichord.github.io/At.js/~~
```js
(function () {
    var textarea = $('#inputor');
    textarea.parent().css('display', 'table');
    textarea.val(Array(300) + ' :');
    textarea.scrollTop(3);
    textarea.focus().atwho('run');
})();
```

__The above code is a incorrect, the scroll event from the `scrollTop` call might trigger the scroll event after the `atwho('run')`. Below is a better version__

```js
(function () {
    var textarea = $('#inputor');
    textarea.parent().css('display', 'table');
    textarea.val(Array(300) + ' :');
    textarea.scrollTop(3);
    setTimeout(function () {
        textarea.focus().atwho('run');
    }, 0);
})();
```

On Chrome (42.0.2311.135): the menu shows up
On Firefox (38.0.5): nothing is shown

I investigated and found out that this breakage is caused by the way Caret.js works and what looks to be a Firefox bug.

Caret.js inserts a dummy element adjacent to the textarea for calculating the caret position. This works fine normally, but for some reason in Firefox, this cause a scroll event to be fired on the textarea. (!!!)
At.js hides the menu on scroll events, which gives the impression that the menu is never shown at all.

The scroll event fires like this if
  - the textarea's parent is `display: table`
  - the textarea is not scrolled to the very top

There really aren't that much At.js and Caret.js can do to fix this perfectly (beside rewriting Caret.js to not insert any elements). I've submitted a [bug report](https://bugzilla.mozilla.org/show_bug.cgi?id=1175351) to Mozilla and hopefully this will be fixed in the future.

~~For now, I propose offering an option that allows turning off "hide-on-scroll" feature to mitigate this issue. I will send a PR for the option soon.~~

nvm I found a hack for fixing it. #308

### PR DESCRIPTION
**Pull request by XrXr**
See #305 

The test case I've added is fairly limited. It doesn't test the handling of the very first scroll event. However I've tested it manually and it does work as expected.

Also, since the bug is Firefox only, it can not be reproduced in PhantomJS. I suggest running the code provided in #305 to verify that this PR fixed the issue.

One additional thing to test is triggering At.js by typing in the textarea produced by the test code. With this PR it should work fine and without it nothing should show up

Closes #305

### CODE REVIEW

**Comment by XrXr** [comment]
Also btw, the last release (731fd924610b71e51f79ffe4bed8baf1299ec208) you did generated a different `jquery.atwho.min.js` which is strange. I suspect it's because you have an outdated npm dependency. Please make sure you run `npm update` next time :)

File: src/app.coffee [hunk 1]
```diff
@@ -79,14 +79,22 @@ class App
...
+      .on('scroll.atwhoInner', ( =>
+        # make returned handler handle the very first call properly
+        lastScrollTop = @$inputor.scrollTop()
+        (e) =>
+          currentScrollTop = e.target.scrollTop
+          if lastScrollTop != currentScrollTop
+            @controller()?.view.hide(e)
+          lastScrollTop = currentScrollTop
+          true  # ensure we don't stop bubbling
+      )())
```
Comment (ichord) [inline src/app.coffee]: how about using `do` ? :

```coffeescript
      .on 'scroll.atwhoInner', do =>
        (e) =>
          currentScrollTop = e.target.scrollTop
```

**Comment by ichord** [comment]
Ooops.. sorry, It's too easy to break PR if I merge another one.  I think it's because of the `min.js` file.

**Comment by XrXr** [comment]
Didn't know about the `do` construct. Thanks!

I've made the change and also rebased.

**Comment by XrXr** [comment]
Can we get a release?

**Comment by ichord** [comment]
@XrXr released.

**Pull request merged**